src/puppetlabs/puppetdb/scf/storage_utils.clj

```diff
@@ -291,17 +291,8 @@
       (catch Exception _
         false))))
 
-(defn async-vacuum-analyze
-  "We don't want to stay in maintenance mode while we vaccum analyze,
-  i.e. refuse http requests as this can take quite some time on large
-  databases"
-  [db-connection-pool]
-  (future
-    (jdbc/with-db-connection db-connection-pool
-      (log/info "Analyzing database")
-      (try
-        (jdbc/do-commands-outside-txn "vacuum (analyze, verbose)")
-        (catch java.sql.SQLException e
-          (log/error e "Caught SQLException during migration")
-          (when-let [next (.getNextException e)]
-            (log/error next "Unravelled exception")))))))
+(defn analyze-small-tables
+  [small-tables]
+  (log/info "Analyzing small tables")
+  (apply jdbc/do-commands-outside-txn
+         (map #(str "analyze " %) small-tables)))
```

Comment:
So *if* we pursued this, I'd wondered if we might just be able to ask for a list of tables, and then ask each table whether it has more than N rows (without any sorting) -- perhaps that'd be fast?  i.e. hand-waving "select count(*) from foo limit 50;"